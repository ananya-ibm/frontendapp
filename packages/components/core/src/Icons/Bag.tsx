/*
Licensed Materials - Property of IBM
694906H
(c) Copyright IBM Corp.  2020 All Rights Reserved

US Government Users Restricted Rights - Use, duplication or disclosure restricted
by GSA ADP Schedule Contract with IBM Corp.
*/

import React from 'react';

export const Bag = ({ width = '20px', fill = 'currentColor' }: Props) => (
  <svg width={width} viewBox="0 0 16 20">
    <g fill="none" fillRule="evenodd">
      <path d="M-4-2h24v24H-4z" />
      <path
        d="M4 4a4 4 0 118 0h3l.8 12.813A3 3 0 0112.808 20H3.193A3 3 0 01.2 16.813L1 4h3zm2 0h4a2 2 0 10-4 0zM2.195 16.938A1 1 0 003.193 18h9.614a1 1 0 00.998-1.062L13.12 6H2.88l-.684 10.938z"
        fill={fill}
      />
    </g>
  </svg>
);

type Props = {
  fill?: string;
  width?: string;
};
