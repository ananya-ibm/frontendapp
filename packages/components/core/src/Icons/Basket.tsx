/*
Licensed Materials - Property of IBM
694906H
(c) Copyright IBM Corp.  2020 All Rights Reserved

US Government Users Restricted Rights - Use, duplication or disclosure restricted
by GSA ADP Schedule Contract with IBM Corp.
*/

import React from 'react';

export const Basket = ({ width = '16px', fill = 'currentColor' }: Props) => (
  <svg width={width} viewBox="0 0 24 21">
    <g stroke="none" strokeWidth="1" fill="none" fillRule="evenodd">
      <g transform="translate(-1322.000000, -50.000000)" fill={fill} fillRule="nonzero">
        <g transform="translate(0.000000, 24.000000)">
          <path d="M1345.4927,33.1544325 C1345.00044,32.5549941 1344.20237,32.1824309 1343.41071,32.1824309 L1340.66286,32.1824309 L1337.16892,26 L1335.49286,26.9476133 L1338.45155,32.1824309 L1329.54661,32.1824309 L1332.5053,26.9476133 L1330.82925,26 L1327.33498,32.1824309 L1324.58456,32.1824309 C1323.79194,32.1824309 1323.02082,32.539591 1322.52182,33.1383876 C1322.09086,33.6547133 1321.91725,34.2884879 1322.03695,34.9437627 L1323.96234,44.3050144 L1323.97902,44.3736867 C1324.39427,45.8379466 1325.31685,46.6783798 1326.50995,46.6783798 L1341.48532,46.6783798 C1342.7718,46.6783798 1343.77333,45.7378263 1344.03261,44.3050144 L1345.95929,34.9370238 C1346.08444,34.3061373 1345.91886,33.6730046 1345.4927,33.1544325 Z M1344.07241,34.5554755 L1342.14252,43.9388692 C1342.10819,44.1298038 1341.95865,44.7523469 1341.48532,44.7523469 L1326.50995,44.7523469 C1326.17012,44.7523469 1325.95961,44.2857605 1325.84088,43.8801447 L1323.92704,34.5766548 C1323.92223,34.5503411 1323.90939,34.4803852 1324.00052,34.3712797 C1324.12985,34.2162857 1324.36988,34.107822 1324.58456,34.107822 L1343.41071,34.107822 C1343.63085,34.107822 1343.87505,34.218532 1344.00502,34.3764141 C1344.08781,34.4771762 1344.07594,34.5368634 1344.07241,34.5554755 Z" />
        </g>
      </g>
    </g>
  </svg>
);

type Props = {
  fill?: string;
  width?: string;
};
