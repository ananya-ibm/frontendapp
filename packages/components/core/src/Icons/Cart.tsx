/*
Licensed Materials - Property of IBM
694906H
(c) Copyright IBM Corp.  2020 All Rights Reserved

US Government Users Restricted Rights - Use, duplication or disclosure restricted
by GSA ADP Schedule Contract with IBM Corp.
*/

import React from 'react';

export const Cart = ({ width = '24px', fill = 'currentColor' }: Props) => (
  <svg width={width} viewBox="0 0 29 24">
    <g stroke="none" strokeWidth="1" fill="none" fillRule="evenodd">
      <g fill={fill}>
        <path d="M11.3684211,22.1052632 C11.3684211,22.6184236 11.1184236,23.0624981 10.6184211,23.4375 C10.1184186,23.8125019 9.52631921,24 8.84210526,24 C8.15789132,24 7.56579197,23.8125019 7.06578947,23.4375 C6.56578697,23.0624981 6.31578947,22.6184236 6.31578947,22.1052632 C6.31578947,21.5921027 6.56578697,21.1480282 7.06578947,20.7730263 C7.56579197,20.3980244 8.15789132,20.2105263 8.84210526,20.2105263 C9.52631921,20.2105263 10.1184186,20.3980244 10.6184211,20.7730263 C11.1184236,21.1480282 11.3684211,21.5921027 11.3684211,22.1052632 Z M27.7894737,22.1052632 C27.7894737,22.6184236 27.5394762,23.0624981 27.0394737,23.4375 C26.5394712,23.8125019 25.9473718,24 25.2631579,24 C24.5789439,24 23.9868446,23.8125019 23.4868421,23.4375 C22.9868396,23.0624981 22.7368421,22.6184236 22.7368421,22.1052632 C22.7368421,21.5921027 22.9868396,21.1480282 23.4868421,20.7730263 C23.9868446,20.3980244 24.5789439,20.2105263 25.2631579,20.2105263 C25.9473718,20.2105263 26.5394712,20.3980244 27.0394737,20.7730263 C27.5394762,21.1480282 27.7894737,21.5921027 27.7894737,22.1052632 Z M29.0526316,3.36842105 L29.0526316,12.3508772 C29.0526316,12.6315804 28.9566052,12.8801159 28.7645496,13.0964912 C28.572494,13.3128666 28.3367929,13.4385963 28.0574393,13.4736842 L9.82970648,15.6140351 C9.98102302,16.315793 10.0566802,16.7251456 10.0566802,16.8421053 C10.0566802,17.0292407 9.91700545,17.403506 9.63765182,17.9649123 L25.7004049,17.9649123 C26.003038,17.9649123 26.264928,18.0760223 26.486083,18.2982456 C26.7072379,18.5204689 26.8178138,18.7836242 26.8178138,19.0877193 C26.8178138,19.3918144 26.7072379,19.6549696 26.486083,19.877193 C26.264928,20.0994163 26.003038,20.2105263 25.7004049,20.2105263 L7.82186235,20.2105263 C7.51922926,20.2105263 7.25733916,20.0994163 7.03618421,19.877193 C6.81502926,19.6549696 6.70445344,19.3918144 6.70445344,19.0877193 C6.70445344,18.9590637 6.75101168,18.774855 6.84412955,18.5350877 C6.93724743,18.2953204 7.03036391,18.0847962 7.12348178,17.9035088 C7.21659966,17.7222213 7.34172492,17.4883055 7.49886134,17.2017544 C7.65599775,16.9152032 7.74620434,16.7426904 7.76948381,16.6842105 L4.6791498,2.24561404 L1.11740891,2.24561404 C0.814775815,2.24561404 0.552885721,2.13450404 0.331730769,1.9122807 C0.110575817,1.69005737 0,1.42690211 0,1.12280702 C0,0.81871193 0.110575817,0.555556667 0.331730769,0.333333333 C0.552885721,0.11111 0.814775815,0 1.11740891,0 L5.58704453,0 C5.77328028,0 5.93914401,0.0380113158 6.08464069,0.114035088 C6.23013737,0.19005886 6.34362307,0.280701228 6.42510121,0.385964912 C6.50657935,0.491228596 6.58223649,0.634502018 6.6520749,0.815789474 C6.7219133,0.99707693 6.76847154,1.14912219 6.79175101,1.27192982 C6.81503048,1.39473746 6.84703927,1.56725035 6.88777834,1.78947368 C6.92851741,2.01169702 6.95470642,2.16374228 6.96634615,2.24561404 L27.9352227,2.24561404 C28.2378558,2.24561404 28.4997459,2.35672404 28.7209008,2.57894737 C28.9420558,2.8011707 29.0526316,3.06432596 29.0526316,3.36842105 Z" />
      </g>
    </g>
  </svg>
);

type Props = {
  fill?: string;
  width?: string;
};
