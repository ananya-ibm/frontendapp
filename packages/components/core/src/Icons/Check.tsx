/*
Licensed Materials - Property of IBM
694906H
(c) Copyright IBM Corp.  2020 All Rights Reserved

US Government Users Restricted Rights - Use, duplication or disclosure restricted
by GSA ADP Schedule Contract with IBM Corp.
*/

import React from 'react';

export const Check = ({ width = '16px', fill = 'currentColor' }: Props) => (
  <svg width={width} viewBox="0 0 16 12">
    <g stroke="none" strokeWidth="1" fill="none" fillRule="evenodd">
      <g transform="translate(-1032.000000, -387.000000)">
        <g transform="translate(844.000000, 293.000000)">
          <g transform="translate(184.000000, 87.000000)">
            <g transform="translate(0.000000, 1.000000)">
              <rect fillRule="nonzero" x="0" y="0" width="24" height="24" />
              <path
                d="M9,15.5857864 L18.2928932,6.29289322 C18.6834175,5.90236893 19.3165825,5.90236893 19.7071068,6.29289322 C20.0976311,6.68341751 20.0976311,7.31658249 19.7071068,7.70710678 L9.70710678,17.7071068 C9.31658249,18.0976311 8.68341751,18.0976311 8.29289322,17.7071068 L4.29289322,13.7071068 C3.90236893,13.3165825 3.90236893,12.6834175 4.29289322,12.2928932 C4.68341751,11.9023689 5.31658249,11.9023689 5.70710678,12.2928932 L9,15.5857864 Z"
                fill={fill}
              />
            </g>
          </g>
        </g>
      </g>
    </g>
  </svg>
);

type Props = {
  fill?: string;
  width?: string;
};
