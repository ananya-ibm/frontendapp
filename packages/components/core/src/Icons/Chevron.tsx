/*
Licensed Materials - Property of IBM
694906H
(c) Copyright IBM Corp.  2020 All Rights Reserved

US Government Users Restricted Rights - Use, duplication or disclosure restricted
by GSA ADP Schedule Contract with IBM Corp.
*/

import React from 'react';

export const Chevron = ({ width = '24px', fill = 'currentColor' }: Props) => (
  <svg width={width} viewBox="0 0 24 24">
    <g stroke="none" strokeWidth="1" fill="none" fillRule="evenodd">
      <g fillRule="nonzero">
        <rect x="0" y="0" width={width} height={width} />
        <path
          d="M14.6829313,12.0000091 L9.24075246,5.65080044 C8.88133031,5.2314746 8.92989185,4.60017461 9.34921769,4.24075246 C9.76854353,3.88133031 10.3998435,3.92989185 10.7592657,4.34921769 L16.7592657,11.3492177 C17.0802569,11.7237074 17.0802569,12.2763107 16.7592657,12.6508004 L10.7592657,19.6508004 C10.3998435,20.0701263 9.76854353,20.1186878 9.34921769,19.7592657 C8.92989185,19.3998435 8.88133031,18.7685435 9.24075246,18.3492177 L14.6829313,12.0000091 Z"
          fill={fill}
        />
      </g>
    </g>
  </svg>
);

type Props = {
  fill?: string;
  width?: string;
};
