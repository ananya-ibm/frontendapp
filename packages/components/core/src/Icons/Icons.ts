/*
Licensed Materials - Property of IBM
694906H
(c) Copyright IBM Corp.  2020 All Rights Reserved

US Government Users Restricted Rights - Use, duplication or disclosure restricted
by GSA ADP Schedule Contract with IBM Corp.
*/

export * from './Bag';
export * from './Basket';
export * from './Cart';
export * from './Check';
export * from './Chevron';
export * from './Pencil';
export * from './Search';
export * from './User';
