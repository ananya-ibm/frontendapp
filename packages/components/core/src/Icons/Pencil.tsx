/*
Licensed Materials - Property of IBM
694906H
(c) Copyright IBM Corp.  2020 All Rights Reserved

US Government Users Restricted Rights - Use, duplication or disclosure restricted
by GSA ADP Schedule Contract with IBM Corp.
*/

import React from 'react';

export const Pencil = ({ width = '16px', fill = 'currentColor' }: Props) => (
  <svg width={width} viewBox="0 0 20 20">
    <g stroke="none" strokeWidth="1" fill="none" fillRule="evenodd">
      <g transform="translate(-407.000000, -477.000000)">
        <g transform="translate(284.000000, 293.000000)">
          <g transform="translate(32.000000, 88.000000)">
            <g transform="translate(89.000000, 94.000000)">
              <g>
                <rect fillRule="nonzero" x="0" y="0" width="24" height="24" />
                <path
                  d="M8.70710678,20.7071068 C8.56750624,20.8467073 8.38970726,20.9418625 8.19611614,20.9805807 L3.19611614,21.9805807 C2.496395,22.1205249 1.8794751,21.503605 2.01941932,20.8038839 L3.01941932,15.8038839 C3.05813755,15.6102927 3.15329267,15.4324938 3.29289322,15.2928932 L15.2928932,3.29289322 C16.787987,1.79779943 19.212013,1.79779943 20.7071068,3.29289322 C22.2022006,4.78798701 22.2022006,7.21201299 20.7071068,8.70710678 L8.70710678,20.7071068 Z M14.999444,6.41357288 L4.92120149,16.4930121 L4.27475488,19.7252451 L7.50698793,19.0787985 L17.585444,8.99957288 L14.999444,6.41357288 Z M16.7071068,4.70710678 L16.413444,4.99957288 L18.999444,7.58557288 L19.2928932,7.29289322 C20.0069384,6.57884801 20.0069384,5.42115199 19.2928932,4.70710678 C18.578848,3.99306157 17.421152,3.99306157 16.7071068,4.70710678 Z"
                  fill={fill}
                  fillRule="nonzero"
                />
              </g>
            </g>
          </g>
        </g>
      </g>
    </g>
  </svg>
);

type Props = {
  fill?: string;
  width?: string;
};
