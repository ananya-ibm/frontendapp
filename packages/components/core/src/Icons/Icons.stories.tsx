/*
Licensed Materials - Property of IBM
694906H
(c) Copyright IBM Corp.  2020 All Rights Reserved

US Government Users Restricted Rights - Use, duplication or disclosure restricted
by GSA ADP Schedule Contract with IBM Corp.
*/

import React from 'react';
import * as Icons from './Icons';

export default { 
  title: 'Components/Core/Icons'
};

export const normal = () => (
  <div>
    {Object.keys(Icons).map(icon => {
      const Icon = Icons[icon];
      return (
        <div
          key={icon}
          style={{
            display: 'inline-block',
            margin: '2rem 0 0 2rem',
            padding: '2rem',
            textAlign: 'center',
            width: '8.75rem'
          }}
        >
          <Icon fill="#000" />
          <div
            style={{
              margin: '2rem auto 0'
            }}
          >
            {icon}
          </div>
        </div>
      );
    })}
  </div>
);
