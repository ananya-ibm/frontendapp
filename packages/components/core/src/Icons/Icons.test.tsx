/*
Licensed Materials - Property of IBM
694906H
(c) Copyright IBM Corp.  2020 All Rights Reserved

US Government Users Restricted Rights - Use, duplication or disclosure restricted
by GSA ADP Schedule Contract with IBM Corp.
*/

/* eslint-disable array-callback-return */
import React from 'react';
import { render } from '@testing-library/react';
import { axe } from 'jest-axe';
import * as Icons from './Icons';

const testProps = {
  color: '#000',
  width: '24px'
};

Object.values(Icons).map(Icon => {
  describe('<Icon /> component', () => {
    test('its snapshot matches', () => {
      const { container } = render(<Icon {...testProps} />);

      expect(container.firstChild).toMatchSnapshot();
    });
    test('it passes an axe accesibility check', async () => {
      const { container } = render(<Icon {...testProps} />);
      const results = await axe(container);
      expect(results).toHaveNoViolations();
    });
  });
});
