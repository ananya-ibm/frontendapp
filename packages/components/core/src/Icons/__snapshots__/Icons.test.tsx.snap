// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Icon /> component its snapshot matches 1`] = `
<svg
  viewBox="0 0 16 20"
  width="24px"
>
  <g
    fill="none"
    fill-rule="evenodd"
  >
    <path
      d="M-4-2h24v24H-4z"
    />
    <path
      d="M4 4a4 4 0 118 0h3l.8 12.813A3 3 0 0112.808 20H3.193A3 3 0 01.2 16.813L1 4h3zm2 0h4a2 2 0 10-4 0zM2.195 16.938A1 1 0 003.193 18h9.614a1 1 0 00.998-1.062L13.12 6H2.88l-.684 10.938z"
      fill="currentColor"
    />
  </g>
</svg>
`;

exports[`<Icon /> component its snapshot matches 2`] = `
<svg
  viewBox="0 0 24 21"
  width="24px"
>
  <g
    fill="none"
    fill-rule="evenodd"
    stroke="none"
    stroke-width="1"
  >
    <g
      fill="currentColor"
      fill-rule="nonzero"
      transform="translate(-1322.000000, -50.000000)"
    >
      <g
        transform="translate(0.000000, 24.000000)"
      >
        <path
          d="M1345.4927,33.1544325 C1345.00044,32.5549941 1344.20237,32.1824309 1343.41071,32.1824309 L1340.66286,32.1824309 L1337.16892,26 L1335.49286,26.9476133 L1338.45155,32.1824309 L1329.54661,32.1824309 L1332.5053,26.9476133 L1330.82925,26 L1327.33498,32.1824309 L1324.58456,32.1824309 C1323.79194,32.1824309 1323.02082,32.539591 1322.52182,33.1383876 C1322.09086,33.6547133 1321.91725,34.2884879 1322.03695,34.9437627 L1323.96234,44.3050144 L1323.97902,44.3736867 C1324.39427,45.8379466 1325.31685,46.6783798 1326.50995,46.6783798 L1341.48532,46.6783798 C1342.7718,46.6783798 1343.77333,45.7378263 1344.03261,44.3050144 L1345.95929,34.9370238 C1346.08444,34.3061373 1345.91886,33.6730046 1345.4927,33.1544325 Z M1344.07241,34.5554755 L1342.14252,43.9388692 C1342.10819,44.1298038 1341.95865,44.7523469 1341.48532,44.7523469 L1326.50995,44.7523469 C1326.17012,44.7523469 1325.95961,44.2857605 1325.84088,43.8801447 L1323.92704,34.5766548 C1323.92223,34.5503411 1323.90939,34.4803852 1324.00052,34.3712797 C1324.12985,34.2162857 1324.36988,34.107822 1324.58456,34.107822 L1343.41071,34.107822 C1343.63085,34.107822 1343.87505,34.218532 1344.00502,34.3764141 C1344.08781,34.4771762 1344.07594,34.5368634 1344.07241,34.5554755 Z"
        />
      </g>
    </g>
  </g>
</svg>
`;

exports[`<Icon /> component its snapshot matches 3`] = `
<svg
  viewBox="0 0 29 24"
  width="24px"
>
  <g
    fill="none"
    fill-rule="evenodd"
    stroke="none"
    stroke-width="1"
  >
    <g
      fill="currentColor"
    >
      <path
        d="M11.3684211,22.1052632 C11.3684211,22.6184236 11.1184236,23.0624981 10.6184211,23.4375 C10.1184186,23.8125019 9.52631921,24 8.84210526,24 C8.15789132,24 7.56579197,23.8125019 7.06578947,23.4375 C6.56578697,23.0624981 6.31578947,22.6184236 6.31578947,22.1052632 C6.31578947,21.5921027 6.56578697,21.1480282 7.06578947,20.7730263 C7.56579197,20.3980244 8.15789132,20.2105263 8.84210526,20.2105263 C9.52631921,20.2105263 10.1184186,20.3980244 10.6184211,20.7730263 C11.1184236,21.1480282 11.3684211,21.5921027 11.3684211,22.1052632 Z M27.7894737,22.1052632 C27.7894737,22.6184236 27.5394762,23.0624981 27.0394737,23.4375 C26.5394712,23.8125019 25.9473718,24 25.2631579,24 C24.5789439,24 23.9868446,23.8125019 23.4868421,23.4375 C22.9868396,23.0624981 22.7368421,22.6184236 22.7368421,22.1052632 C22.7368421,21.5921027 22.9868396,21.1480282 23.4868421,20.7730263 C23.9868446,20.3980244 24.5789439,20.2105263 25.2631579,20.2105263 C25.9473718,20.2105263 26.5394712,20.3980244 27.0394737,20.7730263 C27.5394762,21.1480282 27.7894737,21.5921027 27.7894737,22.1052632 Z M29.0526316,3.36842105 L29.0526316,12.3508772 C29.0526316,12.6315804 28.9566052,12.8801159 28.7645496,13.0964912 C28.572494,13.3128666 28.3367929,13.4385963 28.0574393,13.4736842 L9.82970648,15.6140351 C9.98102302,16.315793 10.0566802,16.7251456 10.0566802,16.8421053 C10.0566802,17.0292407 9.91700545,17.403506 9.63765182,17.9649123 L25.7004049,17.9649123 C26.003038,17.9649123 26.264928,18.0760223 26.486083,18.2982456 C26.7072379,18.5204689 26.8178138,18.7836242 26.8178138,19.0877193 C26.8178138,19.3918144 26.7072379,19.6549696 26.486083,19.877193 C26.264928,20.0994163 26.003038,20.2105263 25.7004049,20.2105263 L7.82186235,20.2105263 C7.51922926,20.2105263 7.25733916,20.0994163 7.03618421,19.877193 C6.81502926,19.6549696 6.70445344,19.3918144 6.70445344,19.0877193 C6.70445344,18.9590637 6.75101168,18.774855 6.84412955,18.5350877 C6.93724743,18.2953204 7.03036391,18.0847962 7.12348178,17.9035088 C7.21659966,17.7222213 7.34172492,17.4883055 7.49886134,17.2017544 C7.65599775,16.9152032 7.74620434,16.7426904 7.76948381,16.6842105 L4.6791498,2.24561404 L1.11740891,2.24561404 C0.814775815,2.24561404 0.552885721,2.13450404 0.331730769,1.9122807 C0.110575817,1.69005737 0,1.42690211 0,1.12280702 C0,0.81871193 0.110575817,0.555556667 0.331730769,0.333333333 C0.552885721,0.11111 0.814775815,0 1.11740891,0 L5.58704453,0 C5.77328028,0 5.93914401,0.0380113158 6.08464069,0.114035088 C6.23013737,0.19005886 6.34362307,0.280701228 6.42510121,0.385964912 C6.50657935,0.491228596 6.58223649,0.634502018 6.6520749,0.815789474 C6.7219133,0.99707693 6.76847154,1.14912219 6.79175101,1.27192982 C6.81503048,1.39473746 6.84703927,1.56725035 6.88777834,1.78947368 C6.92851741,2.01169702 6.95470642,2.16374228 6.96634615,2.24561404 L27.9352227,2.24561404 C28.2378558,2.24561404 28.4997459,2.35672404 28.7209008,2.57894737 C28.9420558,2.8011707 29.0526316,3.06432596 29.0526316,3.36842105 Z"
      />
    </g>
  </g>
</svg>
`;

exports[`<Icon /> component its snapshot matches 4`] = `
<svg
  viewBox="0 0 16 12"
  width="24px"
>
  <g
    fill="none"
    fill-rule="evenodd"
    stroke="none"
    stroke-width="1"
  >
    <g
      transform="translate(-1032.000000, -387.000000)"
    >
      <g
        transform="translate(844.000000, 293.000000)"
      >
        <g
          transform="translate(184.000000, 87.000000)"
        >
          <g
            transform="translate(0.000000, 1.000000)"
          >
            <rect
              fill-rule="nonzero"
              height="24"
              width="24"
              x="0"
              y="0"
            />
            <path
              d="M9,15.5857864 L18.2928932,6.29289322 C18.6834175,5.90236893 19.3165825,5.90236893 19.7071068,6.29289322 C20.0976311,6.68341751 20.0976311,7.31658249 19.7071068,7.70710678 L9.70710678,17.7071068 C9.31658249,18.0976311 8.68341751,18.0976311 8.29289322,17.7071068 L4.29289322,13.7071068 C3.90236893,13.3165825 3.90236893,12.6834175 4.29289322,12.2928932 C4.68341751,11.9023689 5.31658249,11.9023689 5.70710678,12.2928932 L9,15.5857864 Z"
              fill="currentColor"
            />
          </g>
        </g>
      </g>
    </g>
  </g>
</svg>
`;

exports[`<Icon /> component its snapshot matches 5`] = `
<svg
  viewBox="0 0 24 24"
  width="24px"
>
  <g
    fill="none"
    fill-rule="evenodd"
    stroke="none"
    stroke-width="1"
  >
    <g
      fill-rule="nonzero"
    >
      <rect
        height="24px"
        width="24px"
        x="0"
        y="0"
      />
      <path
        d="M14.6829313,12.0000091 L9.24075246,5.65080044 C8.88133031,5.2314746 8.92989185,4.60017461 9.34921769,4.24075246 C9.76854353,3.88133031 10.3998435,3.92989185 10.7592657,4.34921769 L16.7592657,11.3492177 C17.0802569,11.7237074 17.0802569,12.2763107 16.7592657,12.6508004 L10.7592657,19.6508004 C10.3998435,20.0701263 9.76854353,20.1186878 9.34921769,19.7592657 C8.92989185,19.3998435 8.88133031,18.7685435 9.24075246,18.3492177 L14.6829313,12.0000091 Z"
        fill="currentColor"
      />
    </g>
  </g>
</svg>
`;

exports[`<Icon /> component its snapshot matches 6`] = `
<svg
  viewBox="0 0 20 20"
  width="24px"
>
  <g
    fill="none"
    fill-rule="evenodd"
    stroke="none"
    stroke-width="1"
  >
    <g
      transform="translate(-407.000000, -477.000000)"
    >
      <g
        transform="translate(284.000000, 293.000000)"
      >
        <g
          transform="translate(32.000000, 88.000000)"
        >
          <g
            transform="translate(89.000000, 94.000000)"
          >
            <g>
              <rect
                fill-rule="nonzero"
                height="24"
                width="24"
                x="0"
                y="0"
              />
              <path
                d="M8.70710678,20.7071068 C8.56750624,20.8467073 8.38970726,20.9418625 8.19611614,20.9805807 L3.19611614,21.9805807 C2.496395,22.1205249 1.8794751,21.503605 2.01941932,20.8038839 L3.01941932,15.8038839 C3.05813755,15.6102927 3.15329267,15.4324938 3.29289322,15.2928932 L15.2928932,3.29289322 C16.787987,1.79779943 19.212013,1.79779943 20.7071068,3.29289322 C22.2022006,4.78798701 22.2022006,7.21201299 20.7071068,8.70710678 L8.70710678,20.7071068 Z M14.999444,6.41357288 L4.92120149,16.4930121 L4.27475488,19.7252451 L7.50698793,19.0787985 L17.585444,8.99957288 L14.999444,6.41357288 Z M16.7071068,4.70710678 L16.413444,4.99957288 L18.999444,7.58557288 L19.2928932,7.29289322 C20.0069384,6.57884801 20.0069384,5.42115199 19.2928932,4.70710678 C18.578848,3.99306157 17.421152,3.99306157 16.7071068,4.70710678 Z"
                fill="currentColor"
                fill-rule="nonzero"
              />
            </g>
          </g>
        </g>
      </g>
    </g>
  </g>
</svg>
`;

exports[`<Icon /> component its snapshot matches 7`] = `
<svg
  fill="none"
  stroke="currentColor"
  stroke-linecap="round"
  stroke-width="4"
  viewBox="0 0 50 50"
  width="24px"
  xmlns="http://www.w3.org/2000/svg"
>
  <circle
    cx="20"
    cy="20"
    r="15"
  />
  <path
    d="M45 45L31 31"
  />
</svg>
`;

exports[`<Icon /> component its snapshot matches 8`] = `
<svg
  viewBox="0 0 16 16"
  width="24px"
>
  <defs>
    <path
      d="M12,4 C9.79,4 8,5.79 8,8 C8,10.21 9.79,12 12,12 C14.21,12 16,10.21 16,8 C16,5.79 14.21,4 12,4 L12,4 Z M12,14 C9.33,14 4,15.34 4,18 L4,20 L20,20 L20,18 C20,15.33 14.67,14 12,14 L12,14 Z"
      id="user-path"
    />
  </defs>
  <g
    transform="translate(-4.000000, -4.000000)"
  >
    <mask
      fill="white"
    >
      <use
        xlink:href="#user-path"
      />
    </mask>
    <use
      fill="#000"
      xlink:href="#user-path"
    />
  </g>
</svg>
`;
